statsmodels/graphics/functional.py

```diff
@@ -1,14 +1,464 @@
+
+    # Find mean, outliers curves
+    if have_de_optim:
+        median = differential_evolution(lambda x: - ks_gaussian.pdf(x),
+                                        bounds=bounds, maxiter=5).x
+    else:
+        median = brute(lambda x: - ks_gaussian.pdf(x),
+                       ranges=bounds, finish=fmin)
+
+    outliers_idx = np.where(pdf_r < pvalues[alpha.index(threshold)])[0]
+    if labels is not None:
+        labels_outlier = [labels[i] for i in outliers_idx]
+    else:
+        labels_outlier = None
+    outliers = data[outliers_idx]
+
+    # Find HDR given some quantiles
+
+    def _band_quantiles(band):
+        """Find extreme curves for a quantile band.
+
+        From the `band` of quantiles, the associated PDF extrema values
+        are computed. If `min_alpha` is not provided (single quantile value),
+        `max_pdf` is set to `1E6` in order not to constrain the problem on high
+        values.
+
+        An optimization is performed per component in order to find the min and
+        max curves. This is done by comparing the PDF value of a given curve
+        with the band PDF.
+
+        Parameters
+        ----------
+        band : array_like
+            alpha values ``(max_alpha, min_alpha)`` ex: ``[0.9, 0.5]``
+
+        Returns
+        -------
+        band_quantiles : list of 1-D array
+            ``(max_quantile, min_quantile)`` (2, n_features)
+
+        """
+        min_pdf = pvalues[alpha.index(band[0])]
+        try:
+            max_pdf = pvalues[alpha.index(band[1])]
+        except IndexError:
+            max_pdf = 1E6
+        band = [min_pdf, max_pdf]
+
+        pool = Pool()
+        data = zip(range(dim), itertools.repeat((band, pca,
+                                                 bounds, ks_gaussian)))
+        band_quantiles = pool.map(_min_max_band, data)
+        pool.terminate()
+        pool.close()
+
+        band_quantiles = list(zip(*band_quantiles))
+
+        return band_quantiles
+
+    extra_alpha = [i for i in alpha
+                   if 0.5 != i and 0.9 != i and threshold != i]
+    if extra_alpha != []:
+            extra_quantiles = [y for x in extra_alpha
+                               for y in _band_quantiles([x])]
+    else:
+        extra_quantiles = []
+
+    # Inverse transform from n-variate plot to dataset dataset's shape
+    median = _inverse_transform(pca, median)[0]
+    hdr_90 = _band_quantiles([0.9, 0.5])
+    hdr_50 = _band_quantiles([0.5])
+
+    hdr_res = HdrResults({
+                            "median": median,
+                            "hdr_50": hdr_50,
+                            "hdr_90": hdr_90,
+                            "extra_quantiles": extra_quantiles,
+                            "outliers": outliers,
+                            "outliers_idx": outliers_idx
+                         })
+
+    # Plots
+    ax.plot(np.array([xdata] * n_samples).T, data.T,
+            c='c', alpha=.1, label='dataset')
```

Comment:
Do we ever want to have these labels?
When I reformat the legend to 2-columns `ax.legend(ncol=2)` (as I had done in my previous review), then I get a very large number of "dataset" in the legend, maybe one for each year.

Reply:
Indeed, these can be described in the figure's caption. Will remove it.